src/ActionModal/components/ActionModal.vue

```diff
@@ -511,7 +511,9 @@ export default {
       const feeDenom = this.selectedDenom || this.network.stakingDenom
       let balance = this.balances.find(({ denom }) => denom === feeDenom)
       if (!balance) {
-        balance = defaultBalance
+        // HACK for now. If not balance we hard reload
```

Comment:
why not return an empty balance here? this gets reevaluated once this.balances updates, right?

Reply:
I was debugging this and this is the only solution I could find for now... I don't know why it happens sometimes. You get the right balances, and then in the feeStep the old ones show up again... it is very strange

Reply:
but with this hack it is safe

Reply:
but it forces a refresh, it's a flag for me

Reply:
How can I reproduce this?

Reply:
This will also refresh the page if the user has no balance in that denom.

Reply:
If it has no balance it cannot be selected in the dropdown in the first place (it doesn't appear)

Reply:
To reproduce: you need to have your accounts signed in in at least two or three networks (I have sessions in all networks, so idk). Then you navigate to portfolio, make a transaction. Navigate to portfolio to another account. Try to make a transaction. You will probably be greeted with a "You don't have enough tokens to proceed" even though you have enough.

The reason is because in selectedBalance it is searching for a denom that only exists in another network

Reply:
So something is not resetting correctly when closing/opening the actionmodal? Need to wait for some queries to finish?

Reply:
🤔 probably something like that, like reopening the modal doesn't refresh the values... 

But still, we don't need the balances query there. Getting it as a prop from `SendModal` and `ModalWithdrawRewards` covers it all

Reply:
passing the balances around creates unwanted dependencies between the parent and child components in my opinion

Reply:
which already exists. We are already sending couple of required props from those "children" (see rewards)

Reply:
but yeah, I get your point. You are concerned about going to the root of the thing: why those balances are not updating in ActionModal (but they do in SendModal)

Reply:
can we revert this change? as the network switching issue seems to come from somewhere else?

Reply:
Ok, let me revert it and put balances back there as an apollo query. But we need to fix this asap, because for users it is annoying (they need to refresh to perform transactions and they will be confused by wrong validation)

Reply:
Done